File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetsbillpaymentpagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel } from "./codatdatacontractsdatasetsbillpaymentpagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsBillPaymentPagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel)
  current?: CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel)
  next?: CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel)
  self?: CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel;
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_billpaymentpagedresponselinksmodel.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import codat_datacontracts_datasets_billpaymentpagedresponsehrefmodel as shared_codat_datacontracts_datasets_billpaymentpagedresponsehrefmodel


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsBillPaymentPagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_billpaymentpagedresponsehrefmodel.CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current') }})
    next: Optional[shared_codat_datacontracts_datasets_billpaymentpagedresponsehrefmodel.CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    previous: Optional[shared_codat_datacontracts_datasets_billpaymentpagedresponsehrefmodel.CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous') }})
    self: Optional[shared_codat_datacontracts_datasets_billpaymentpagedresponsehrefmodel.CodatDataContractsDatasetsBillPaymentPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self') }})
    
